internal/server/receiver_handler_test.go

```diff
@@ -265,8 +298,234 @@ func Test_validate(t *testing.T) {
+				},
+			},
+			expectedResponseCode: http.StatusBadRequest,
+		},
+		{
+			name:                 "no receiver configured",
+			expectedResponseCode: http.StatusNotFound,
+		},
+		{
+			name: "not ready receiver is ignored",
+			receiver: &apiv1.Receiver{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "notready-receiver",
+				},
+				Spec: apiv1.ReceiverSpec{},
+				Status: apiv1.ReceiverStatus{
+					WebhookPath: apiv1.ReceiverWebhookPath,
+					Conditions:  []metav1.Condition{{Type: meta.StalledCondition, Status: metav1.ConditionFalse}},
+				},
+			},
+			expectedResponseCode: http.StatusNotFound,
+		},
+		{
+			name: "suspended receiver ignored",
+			receiver: &apiv1.Receiver{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "suspended-receiver",
+				},
+				Spec: apiv1.ReceiverSpec{
+					Suspend: true,
+				},
+				Status: apiv1.ReceiverStatus{
+					WebhookPath: apiv1.ReceiverWebhookPath,
+					Conditions:  []metav1.Condition{{Type: meta.ReadyCondition, Status: metav1.ConditionTrue}},
+				},
+			},
+			expectedResponseCode: http.StatusNotFound,
+		},
+		{
+			name: "missing apiVersion in resource",
+			receiver: &apiv1.Receiver{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "receiver",
+				},
+				Spec: apiv1.ReceiverSpec{
+					Type: apiv1.GenericReceiver,
+					SecretRef: meta.LocalObjectReference{
+						Name: "token",
+					},
+					Resources: []apiv1.CrossNamespaceObjectReference{
+						{
+							Kind: apiv1.ReceiverKind,
+							MatchLabels: map[string]string{
+								"label": "match",
+							},
+						},
+					},
+				},
+				Status: apiv1.ReceiverStatus{
+					WebhookPath: apiv1.ReceiverWebhookPath,
+					Conditions:  []metav1.Condition{{Type: meta.ReadyCondition, Status: metav1.ConditionTrue}},
+				},
+			},
+			secret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "token",
+				},
+				Data: map[string][]byte{
+					"token": []byte("token"),
+				},
+			},
+			expectedResponseCode: http.StatusBadRequest,
+		},
+		{
+			name: "resource by name not found",
+			receiver: &apiv1.Receiver{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "receiver",
+				},
+				Spec: apiv1.ReceiverSpec{
+					Type: apiv1.GenericReceiver,
+					SecretRef: meta.LocalObjectReference{
+						Name: "token",
+					},
+					Resources: []apiv1.CrossNamespaceObjectReference{
+						{
+							APIVersion: apiv1.GroupVersion.String(),
+							Kind:       apiv1.ReceiverKind,
+							Name:       "does-not-exists",
+						},
+					},
+				},
+				Status: apiv1.ReceiverStatus{
+					WebhookPath: apiv1.ReceiverWebhookPath,
+					Conditions:  []metav1.Condition{{Type: meta.ReadyCondition, Status: metav1.ConditionTrue}},
+				},
+			},
+			secret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "token",
+				},
+				Data: map[string][]byte{
+					"token": []byte("token"),
+				},
+			},
+			expectedResponseCode: http.StatusBadRequest,
+		},
+		{
+			name: "annotating resources by label match",
```

Comment:
please add tests for the edge cases:

- Name == '*' and MatchLables is nil/empty
- Name != '*' and != '' and MatchLabels is present

Reply:
> Name == '*' and MatchLables is nil/empty
@makkes there is still that open question if name is `*` and not matchLabels are defined if it should target all resources matching the kind? I would opt for yes.

> Name != '*' and != '' and MatchLabels is present
In that case we agree matchLabel would just be ignored right? And It would just get the resource by name.

Reply:
For Alerts if Name == '*' then it matches any source so I agree we should implement the same behaviour here as well.

Also for Alerts if Name != '*' and MatchLabels != nil then, yes, MatchLabels is just ignored and we should replicate that behaviour.